src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useState } from 'react';
+import { Alert } from '@mui/material';
+import useFeatureStrategyApi from 'hooks/api/actions/useFeatureStrategyApi/useFeatureStrategyApi';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import useToast from 'hooks/useToast';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { StrategyDraggableItem } from './StrategyDraggableItem/StrategyDraggableItem';
+import { IFeatureEnvironment } from 'interfaces/featureToggle';
+import { FeatureStrategyEmpty } from 'component/feature/FeatureStrategy/FeatureStrategyEmpty/FeatureStrategyEmpty';
+import { useRequiredPathParam } from 'hooks/useRequiredPathParam';
+import { useStyles } from './EnvironmentAccordionBody.styles';
+import { useFeature } from 'hooks/api/getters/useFeature/useFeature';
+
+interface IEnvironmentAccordionBodyProps {
+    isDisabled: boolean;
+    featureEnvironment?: IFeatureEnvironment;
+}
+
+const EnvironmentAccordionBody = ({
+    featureEnvironment,
+    isDisabled,
+}: IEnvironmentAccordionBodyProps) => {
+    const projectId = useRequiredPathParam('projectId');
+    const featureId = useRequiredPathParam('featureId');
+    const { setStrategiesSortOrder } = useFeatureStrategyApi();
+    const { setToastData, setToastApiError } = useToast();
+    const { refetchFeature } = useFeature(projectId, featureId);
+
+    const [strategies, setStrategies] = useState(
+        featureEnvironment?.strategies || []
+    );
+    const { classes: styles } = useStyles();
+    useEffect(() => {
+        // Use state to enable drag and drop, but switch to API output when it arrives
+        setStrategies(featureEnvironment?.strategies || []);
+    }, [featureEnvironment?.strategies]);
+
+    if (!featureEnvironment) {
+        return null;
+    }
+
+    const onDragAndDrop = async (
+        from: number,
+        to: number,
+        dropped?: boolean
+    ) => {
+        if (from !== to && dropped) {
+            const newStrategies = [...strategies];
+            const movedStrategy = newStrategies.splice(from, 1)[0];
```

Comment:
Is this accessor `[0]` guaranteed to succeed?

Reply:
in `from !== to`, yes. I reason that the fact it was moved to a different index implies that at least 2 items exist.